exchange_machine.lua

```diff
@@ -87,16 +77,18 @@ local tablecolumns =
 
 
 local function table_from_results(results, x, y, w, h, selected)
-	local fs_tab = {}
+	local fs_tab
 
 	local function insert(str)
-		table.insert(fs_tab, str)
+		fs_tab[#fs_tab+1] = str
```

Comment:
table.insert causes a table lookup, function call to lua, which then does the work

the assignment to [#num + 1] avoids all of that.